framework/src/controls/WebServerControl.C

```diff
@@ -364,3 +436,62 @@ WebServerControl::stringifyJSONType(const miniJson::JsonType & json_type)
     return "object";
   ::mooseError("WebServerControl::stringifyJSONType(): Unused JSON value type");
 }
+
+template <>
+miniJson::Json
+WebServerControl::toMiniJson(const nlohmann::json & value)
+{
+  const auto value_str = value.dump();
+  std::string errMsg;
+  const auto json_value = miniJson::Json::parse(value_str, errMsg);
+  if (!errMsg.empty())
+    ::mooseError("Failed parse value into miniJson:\n", errMsg);
+  return json_value;
+}
+
+WebServerControl::RealEigenMatrixValue::RealEigenMatrixValue(const std::string & name,
+                                                             const std::string & type)
+  : TypedValueBase<RealEigenMatrix>(name, type)
+{
+}
+
+WebServerControl::RealEigenMatrixValue::RealEigenMatrixValue(const std::string & name,
+                                                             const std::string & type,
+                                                             const miniJson::Json & json_value)
+  : TypedValueBase<RealEigenMatrix>(name, type, getMatrixJSONValue(json_value))
+{
+}
+
+RealEigenMatrix
+WebServerControl::RealEigenMatrixValue::getMatrixJSONValue(const miniJson::Json & json_value)
+{
+  const auto from_json_type = json_value.getType();
+  if (from_json_type != miniJson::JsonType::kArray)
+    throw ValueBase::Exception("The value '" + json_value.serialize() + "' of type " +
+                               stringifyJSONType(from_json_type) + " is not an array");
+
+  const auto & array_of_array_value = json_value.toArray();
+  const auto nrows = array_of_array_value.size();
+  if (nrows == 0)
+    return RealEigenMatrix::Zero(0, 0);
+
+  RealEigenMatrix matrix;
+  for (const auto i : make_range(nrows))
+  {
+    if (array_of_array_value[i].getType() != miniJson::JsonType::kArray)
+      throw ValueBase::Exception(
+          "The value '" + array_of_array_value[i].serialize() + "' of type " +
```

Comment:
Is it more useful to have the index of these errors instead of the value itself?